src/Catalyst.Core.Lib.Tests/UnitTests/IO/Observers/ResponseObserverBaseTests.cs

```diff
@@ -0,0 +1,82 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.Linq;
+using System.Threading;
+using Catalyst.Abstractions.IO.Messaging.Correlation;
+using Catalyst.Core.Lib.IO.Observers;
+using Catalyst.Protocol.Peer;
+using Catalyst.Protocol.Rpc.Node;
+using Catalyst.TestUtils;
+using DotNetty.Transport.Channels;
+using FluentAssertions;
+using Microsoft.Reactive.Testing;
+using NSubstitute;
+using Serilog;
+using Xunit;
+
+namespace Catalyst.Core.Lib.Tests.UnitTests.IO.Observers
+{
+    public class ResponseObserverBaseTests
+    {
+        private sealed class FailingResponseObserver : ResponseObserverBase<GetPeerCountResponse>
+        {
+            private int _counter;
+            public int Counter => _counter;
+
+            public FailingResponseObserver(ILogger logger) : base(logger) { }
+
+            protected override void HandleResponse(GetPeerCountResponse messageDto,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused private method 'HandleResponse'.](https://app.codacy.com/app/catalyst-network/Catalyst.Node/pullRequest?prid=4318201)